cpp/modmesh/buffer/small_vector.hpp

```diff
@@ -330,42 +333,92 @@ class small_vector
         return mid;
     }
 
+    /**
```

Comment:
Please move the long function (and all long functions you add in this PR) outside the class declaration.

Long function bodies in class declaration make it hard to read.